Source/Core/VideoCommon/VertexLoaderManager.cpp

```diff
@@ -72,16 +72,35 @@ void UpdateVertexArrayPointers()
   if (!g_main_cp_state.bases_dirty)
     return;
 
+  // Only update the array base if the vertex description states we are going to use it.
+  auto is_indexed = [](VertexComponentFormat f) {
```

Comment:
I've seen this multiple places.  Would it make sense to turn this into a free-form function that lives next to the enum?

Reply:
I think that could be nice; the one reason why I've been hesitant to do so is because BP/CP/XFMemory are all in global scope, and adding `IsIndexed` to the global scope seems a bit jank.  (The proper solution would be to refactor them into a namespace, which I guess isn't any harder if a function is added though.)